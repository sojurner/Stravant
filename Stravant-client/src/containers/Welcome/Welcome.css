.fa-sign-in-alt {
  font-size: 3rem;
  cursor: pointer; }

.fa-sign-in-alt:hover {
  color: #826868; }

.welcome-title {
  color: crimson;
  font-family: daybold; }
