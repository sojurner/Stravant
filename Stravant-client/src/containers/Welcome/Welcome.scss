.fa-sign-in-alt {
  font-size: 3rem;
  cursor: pointer;
}

.fa-sign-in-alt:hover {
  color: #826868;
}

.welcome-title {
  color: crimson;
  font-family: daybold;
  margin: 1rem auto;
  font-size: 5rem;
}

.speech-bubble {
  font-family: daybold;
  color: rgb(255, 193, 184);
  letter-spacing: 2px;
  font-size: 1.5rem;
  position: relative;
  background: #c5696a;
  border-radius: 0.4em;
  padding: 1rem;
  opacity: 0.9;
}

.speech-bubble:after {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  width: 0;
  height: 0;
  border: 21px solid transparent;
  border-right-color: #c5696a;
  border-left: 0;
  border-bottom: 0;
  margin-top: -10.5px;
  margin-left: -21px;
  opacity: 0.9;
}
