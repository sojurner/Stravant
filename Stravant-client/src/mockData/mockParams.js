export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];
