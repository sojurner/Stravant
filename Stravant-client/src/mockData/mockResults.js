export const aggregateResult = { Biked: 0, Ran: 6.472964574269732, Swam: 0 };

export const tokenResult = {
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    badge_type_id: 0,
    city: null,
    country: null,
    created_at: '2018-09-04T22:53:48Z',
    email: 'pykim0591@gmail.com',
    firstname: 'Paul',
    follower: null,
    friend: null,
    id: 34515396,
    lastname: 'K',
    premium: false,
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    resource_state: 2,
    sex: 'M',
    state: null,
    summit: false,
    updated_at: '2018-09-07T12:20:05Z',
    username: null
  },
  token_type: 'Bearer'
};

export const weeklyResult = {
  Friday: 3784.3,
  Monday: 0,
  Saturday: 0,
  Sunday: 0,
  Thursday: 0,
  Tuesday: 0,
  Wednesday: 0
};
