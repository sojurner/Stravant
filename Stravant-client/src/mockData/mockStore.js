export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubInfo: { clubName: 'Stravant Club', clubId: 473825 },
  clubActivity: {
    mostRecent: {
      name: 'Pamela',
      distance: 4.270478558110628,
      activity: 'Morning Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    },
    Paul: { totalDistance: 6.46, totalTime: 78, membersRecent: 'Morning Run' },
    Jesse: {
      totalDistance: 277.78000000000003,
      totalTime: 5292,
      membersRecent: 'Morning Run'
    }
  }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};
