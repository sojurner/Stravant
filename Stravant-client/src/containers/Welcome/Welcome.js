import React, { Component } from 'react';
import { connect } from 'react-redux';
import { stravaApi } from '../../data/strava_config';
import * as userActions from '../../actions/userAction';
import * as apiCalls from '../../helpers/apiCalls/apiCalls';
import * as scrape from '../../helpers/helpers/helpers';

import './Welcome.css';
export class Welcome extends Component {
  constructor() {
    super();
    this.state = {
      timer: 0
    };
  }
  componentDidMount() {
    const { search } = window.location;
    const codeIndex = search.indexOf('code') + 5;
    const lastIndex = search.lastIndexOf('&');
    const userCode = search.slice(codeIndex, lastIndex);
    this.exchangeToken(userCode);
  }

  exchangeToken = async code => {
    const result = await apiCalls.exchangeUserToken(code);
    if (!result.message) {
      const scrapedUserInfo = scrape.userInfo(result);
      this.props.setAccessToken(scrapedUserInfo);
    }
    localStorage.setItem('code', JSON.stringify({ code }));
  };

  togglePom = () => {
    const { togglePomState, currentUser, pomStatus } = this.props;
    if (pomStatus) {
      this.setState({ timer: 0 });
    }
    togglePomState(!currentUser.pomStatus);
  };

  startTimer = () => {
    let time = this.state.timer + 1;
    this.setState({ timer: time });
  };

  signOutUser = () => {
    localStorage.removeItem('code');
    window.location = 'http://localhost:3000/';
  };

  render() {
    const { info, pomStatus } = this.props.currentUser;
    return (
      <div className="welcome-page">
        {this.state.timer > 0 && pomStatus && <h4>{this.state.timer}</h4>}
        {this.state.timer > 0 &&
          !pomStatus && (
            <h4>
              You took a {this.state.timer}
              sec pom
            </h4>
          )}
        {info && <h2>HI {info.firstName}</h2>}
        <i className="fas fa-sign-in-alt" onClick={this.signOutUser} />
        {pomStatus ? (
          <button
            onClick={() => {
              this.togglePom();
              clearInterval(this.time);
            }}
          >
            Done!
          </button>
        ) : (
          <button
            onClick={() => {
              this.togglePom();
              this.time = setInterval(this.startTimer, 1000);
            }}
          >
            POM TIME
          </button>
        )}
      </div>
    );
  }
}
const mapStateToProps = state => ({
  currentUser: state.currentUser
});

const mapDispatchToProps = dispatch => ({
  setAccessToken: token => dispatch(userActions.setAccessToken(token)),
  togglePomState: bool => dispatch(userActions.togglePomState(bool)),
  setWeeklyStats: stats => dispatch(userActions.setWeeklyStats(stats))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Welcome);
