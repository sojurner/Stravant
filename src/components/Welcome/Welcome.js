import React, { Component } from 'react';
import { stravaApi } from '../../data/strava_config';

import './Welcome.css';

class Welcome extends Component {
  async componentDidMount() {
    const url = `https://www.strava.com/oauth/authorize/?client_id=${
      stravaApi.client_id
    }&redirect_uri=${stravaApi.redirect_uri}&response_type=code`;
    const options = {
      mode: 'no-cors'
    };
    fetch(url, options).then(res => {
      console.log(res);
    });
  }

  render() {
    const stravant = 'S†ra√an†';
    const url = `https://www.strava.com/oauth/authorize/client_id=${
      stravaApi.client_id
    }&redirect_uri=${stravaApi.redirect_uri}&response_type='code'`;
    return (
      <div className="welcome-page">
        <h1 className="welcome-title">{stravant}</h1>

        <i class="fas fa-sign-in-alt" />
        <a href={url}>login w/strava</a>
      </div>
    );
  }
}
export default Welcome;
