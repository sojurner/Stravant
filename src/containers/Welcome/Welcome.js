import React, { Component } from 'react';
import { connect } from 'react-redux';
import { stravaApi } from '../../data/strava_config';
import * as userActions from '../../actions/userAction';
import { exchangeUserToken } from '../../helpers/apiCalls/apiCalls';

import './Welcome.css';
export class Welcome extends Component {
  componentDidMount() {
    const userCode = window.location.search.slice(13, 53);
    this.exchangeToken(userCode);
  }

  exchangeToken = async code => {
    const result = await exchangeUserToken(code);
    if (!result.message) {
      this.props.setAccessToken(result);
    }
  };

  handleClick = () => {
    const url = `https://www.strava.com/oauth/authorize?client_id=${
      stravaApi.client_id
    }&redirect_uri=${
      stravaApi.redirect_uri
    }&response_type=code&approval_prompt=force`;

    window.location = url;
  };

  togglePom = () => {
    const { togglePomState, currentUser } = this.props;
    togglePomState(!currentUser.pomStatus);
    console.log(currentUser);
  };

  render() {
    const stravant = 'S†ra√an†';
    return (
      <div className="welcome-page">
        <h1 className="welcome-title">{stravant}</h1>
        <i className="fas fa-sign-in-alt" />
        <button onClick={this.handleClick}>login w/strava</button>
        <button onClick={this.togglePom}>POM</button>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  currentUser: state.currentUser
});

const mapDispatchToProps = dispatch => ({
  setAccessToken: token => dispatch(userActions.setAccessToken(token)),
  togglePomState: bool => dispatch(userActions.togglePomState(bool))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Welcome);
