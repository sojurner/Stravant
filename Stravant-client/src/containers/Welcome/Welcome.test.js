import React from 'react';
import { shallow } from 'enzyme';
import { Welcome, mapStateToProps, mapDispatchToProps } from './Welcome';
import * as store from '../../mockData/mockStore';

jest.mock('../../helpers/apiCalls/apiCalls');

describe('Welcome', () => {
  let wrapper;
  let mockCurrentUser;
  let mockSetToken;

  beforeEach(() => {
    localStorage.clear();
    mockSetToken = jest.fn();
    mockCurrentUser = store.currentUser;

    wrapper = shallow(
      <Welcome currentUser={mockCurrentUser} setAccessToken={mockSetToken} />
    );
  });

  it('should match snapshot when user is male', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapShot when user is female', () => {
    mockCurrentUser = {
      info: {
        accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
        firstName: 'Pam',
        gender: 'F',
        profilePic:
          'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
        userId: 34515396
      },
      totalStats: {},
      weeklyStats: {}
    };

    wrapper = shallow(
      <Welcome currentUser={mockCurrentUser} setAccessToken={mockSetToken} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should set an item in localStorage with the url code', async () => {
    const url = '?state=&code=db45796a99ddced0034a049288f7ff5c2fdc10e6&scope=';
    const expected = '{"code":"db45796a99ddced0034a049288f7ff5c2fdc10e6"}';

    await wrapper.instance().exchangeToken(url);
    expect(mockSetToken).toHaveBeenCalled();
    expect(localStorage.store.code).toEqual(expected);
  });

  it('should map to store properly', () => {
    const mockStore = store.currentUser;
    const mapped = mapStateToProps(mockStore);

    expect(mapped.currentUser).toEqual(mockStore.currentUser);
  });

  it('should call dispatch when a function from dispatch is called', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.setAccessToken();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
