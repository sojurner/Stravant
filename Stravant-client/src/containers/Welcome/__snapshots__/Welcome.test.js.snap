// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match snapShot when user is female 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    currentUser={
        Object {
            "info": Object {
              "accessToken": "a7eeea216a46a760a74c60acbffc3b55c66537c5",
              "firstName": "Pam",
              "gender": "F",
              "profilePic": "https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg",
              "userId": 34515396,
            },
            "totalStats": Object {},
            "weeklyStats": Object {},
          }
    }
    setAccessToken={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        <img
                className="avatar"
                height="100"
                src="female-avatar.png"
                width="100"
        />
        <h4
                className="speech-bubble"
        >
                Hi 
                Pam
        </h4>
</span>,
      "className": "welcome-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <img
            className="avatar"
            height="100"
            src="female-avatar.png"
            width="100"
/>,
          <h4
            className="speech-bubble"
>
            Hi 
            Pam
</h4>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "avatar",
            "height": "100",
            "src": "female-avatar.png",
            "width": "100",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hi ",
              "Pam",
            ],
            "className": "speech-bubble",
          },
          "ref": null,
          "rendered": Array [
            "Hi ",
            "Pam",
          ],
          "type": "h4",
        },
      ],
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          <img
                    className="avatar"
                    height="100"
                    src="female-avatar.png"
                    width="100"
          />
          <h4
                    className="speech-bubble"
          >
                    Hi 
                    Pam
          </h4>
</span>,
        "className": "welcome-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <img
              className="avatar"
              height="100"
              src="female-avatar.png"
              width="100"
/>,
            <h4
              className="speech-bubble"
>
              Hi 
              Pam
</h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "avatar",
              "height": "100",
              "src": "female-avatar.png",
              "width": "100",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hi ",
                "Pam",
              ],
              "className": "speech-bubble",
            },
            "ref": null,
            "rendered": Array [
              "Hi ",
              "Pam",
            ],
            "type": "h4",
          },
        ],
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Welcome should match snapshot when user is male 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    currentUser={
        Object {
            "info": Object {
              "accessToken": "a7eeea216a46a760a74c60acbffc3b55c66537c5",
              "firstName": "Paul",
              "gender": "M",
              "profilePic": "https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg",
              "userId": 34515396,
            },
            "totalStats": Object {},
            "weeklyStats": Object {},
          }
    }
    setAccessToken={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        <img
                className="avatar"
                height="70"
                src="male-avatar.png"
                width="70"
        />
        <h4
                className="speech-bubble"
        >
                Hi 
                Paul
        </h4>
</span>,
      "className": "welcome-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="avatar"
            height="70"
            src="male-avatar.png"
            width="70"
/>,
          false,
          <h4
            className="speech-bubble"
>
            Hi 
            Paul
</h4>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "avatar",
            "height": "70",
            "src": "male-avatar.png",
            "width": "70",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hi ",
              "Paul",
            ],
            "className": "speech-bubble",
          },
          "ref": null,
          "rendered": Array [
            "Hi ",
            "Paul",
          ],
          "type": "h4",
        },
      ],
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          <img
                    className="avatar"
                    height="70"
                    src="male-avatar.png"
                    width="70"
          />
          <h4
                    className="speech-bubble"
          >
                    Hi 
                    Paul
          </h4>
</span>,
        "className": "welcome-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="avatar"
              height="70"
              src="male-avatar.png"
              width="70"
/>,
            false,
            <h4
              className="speech-bubble"
>
              Hi 
              Paul
</h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "avatar",
              "height": "70",
              "src": "male-avatar.png",
              "width": "70",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hi ",
                "Paul",
              ],
              "className": "speech-bubble",
            },
            "ref": null,
            "rendered": Array [
              "Hi ",
              "Paul",
            ],
            "type": "h4",
          },
        ],
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
